whatshap/cli/haplotag.py

```diff
@@ -378,8 +381,11 @@ def open_output_alignment_file(aln_output, reference, vcf_md5, bam_header):
             )
         kwargs = dict(mode="wc", reference_filename=reference)
     else:
-        # Write BAM
-        kwargs = dict(mode="wb")
+        # Write BAM, disable compression for stdout
+        if aln_output is sys.stdout:
```

Comment:
I think compression should be only be disabled if stdout is not a regular file. Otherwise, running `whatshap haplotag ... > output.bam` will result in huge files.

Can you do this instead:
```suggestion
        if aln_output is sys.stdout and not stdout_is_regular_file():
```
And then add this somewhere (plus imports):
```
def stdout_is_regular_file() -> bool:
    mode = os.fstat(sys.stdout.buffer.fileno()).st_mode
    return stat.S_ISREG(mode)
```

Reply:
I thought about this. To be more in line with samtools itself I thought of adding an explicit `-u / --uncompressed` option, but thought that a bit clumsy because users would need to know to use the option to yield any benefit from piping (in my use case). Your idea is nicer because it will result in always doing a sensible thing.

I'll implement that.

Reply:
Implemented in [3686cf5](https://github.com/whatshap/whatshap/pull/352/commits/3686cf567ce57b9e1a4744f79968fcfb26cf5464).